app/src/main/java/com/example/parkingsystemkotlin/mvp/model/reservation/ReservationInformationDB.kt

```diff
@@ -65,9 +65,18 @@ object ReservationInformationDB : DataBase {
         return releasedReservations
     }
 
+    override fun isOverlap(startDateAndTime: Calendar, finishDateAndTime: Calendar, place: String) =
+        reservations[place]?.filter { !(startDateAndTime.after(it.finishDateAndTime) || finishDateAndTime.before(it.startDateAndTime)) }?.size ?: 0 > 0
+
+    override fun getAllReservations(): MutableList<Reservation> {
+        val reservationsList: MutableList<Reservation> = mutableListOf()
+        reservations.forEach { reservation -> reservationsList.addAll(reservation.value) }
+        return reservationsList
+    }
+
     private fun deleteReservations(key: String): Int {
         val calendar = Calendar.getInstance(Locale.getDefault())
-        var list: MutableList<Reservation>? = reservations[key]
+        val list: MutableList<Reservation>? = reservations[key]
```

Comment:
why you use a variable if you could use the property of the filter?

Reply:
done